### ISSUE TITLE
SVG and CSS example does not work

### ISSUE THREAD

**Issue by isabeau-bergeron** [mdn/sprints#1987]
# Request type

- [ ] Please close this issue, I accidentally submitted it without adding any details
- [ ] New documentation
- [x] Correction or update

# Details
The example does not work. I tested with latest versions of Mozilla, Chrome and Safari on mac os.

<img width="605" alt="Screen Shot 2019-08-09 at 1 34 18 PM" src="https://user-images.githubusercontent.com/15198944/62797781-87f99c00-baaa-11e9-9267-0d50c8caa61f.png">

**Comment by chrisdavidmills** [comment]
Example found at page https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/SVG_and_CSS

**Comment by SebastianZ** [comment]
TL;DR: I have fixed the example using the standardized way, though this is currently only supported in Firefox.

Background: The styling of elements referenced via `<use>` elements changed between SVG 1.1 and SVG 2. The new way [defines that the referenced elements inherit all styles defined on the `<use>` element](https://svgwg.org/svg2-draft/struct.html#UseStyleInheritance).
This means, to style elements referenced by a `<use>` element individually, you need to use CSS custom properties for them. Then you can apply different values for those properties on the different `<use>` elements.

Obviously, the only browser that currently supports this behavior is Firefox. All other browsers didn't get updated yet.

The issue we are facing in the example is that the `:hover` rules set on the petals, which are referenced via several cascaded `<use>` elements, are only interpreted by Gecko-based browsers.

There is [some discussion about what rules apply to elements referenced by <use> elements](https://github.com/w3c/svgwg/issues/504) in the SVG Working Group, as it seems hard for some browser implementors to apply all rules as if the elements where not referenced via `<use>` elements.

So there are three things we can do now:

1. Keep the example as is for now and follow the discussion around the standardization.
2. Change the example to use several individual elements for the petals instead of referencing the same ones many times via `<use>` elements. Though that would also make the example code much longer.
3. Replace the example by a simpler one, which doesn't use `<use>` elements, or at least without `:hover` styling.

Because it's unclear whether the other browsers will eventually settle down on this behavior, I'd say we may do go with option 2 or 3. Maybe with 2 first, because it's easier to do and then there's time to do option 3.

Sebastian

**Comment by chrisdavidmills** [comment]
Thanks a lot for the updates and research, @SebastianZ ! I agree with your plan about how to update this for now. It might be worth filling in some details on the page to say what the state of standardization is too.

**Comment by SebastianZ** [comment]
I've implemented option 2 and it now works even in good ol' IE 11. 😃 As I said, the code is much longer now, though, so we should try to create a simpler example at some point.

I have kept the more concise example using `<use>` elements, only moved it down, and added a short description regardng the state of standardization.

Sebastian
